api/src/main/java/jakarta/data/spi/provider/package-info.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * <p>This package defines the Service Provider Interface (SPI) by which
+ * Jakarta Data providers can register to provide repository instances
+ * to the Jakarta EE product's CDI extension that handles
```

Comment:
@njr-11 @Emily-Jiang @frowe 
That is what I'm not following.

The reason to need to register to Jakarta Data and CDI.

We agree that should be CDI-centric; my point is we don't need this SPI; the CDI extension is complete.

E.g.:

You can find the repositories in the `BeforeBeanDiscovery`, and using the event `AfterBeanDiscovery ` you can register the repositories to CDI.

This interface has an [add bean](https://jakarta.ee/specifications/cdi/3.0/apidocs/) method to do without this SPI.

Why not use the CDI capabilities to make it happen?

Why should I make this mandatory if I can make it easier only using CDI?


If you take a look at Bean from CDI is exactly what you're looking for: 
https://jakarta.ee/specifications/cdi/4.0/jakarta-cdi-spec-4.0.html#bean

My point is, CDI provides this discovery and it works like a charm.

Reply:
I'm pretty sure that Spring and Micronaut have not ever agreed to have their Jakarta Data providers be CDI-centric.  Even if they did, what you suggest would cause collisions in a Jakarta EE profile where multiple providers all try to handle the injection point with different CDI extensions.  What you are suggesting only works like a charm in a Standalone mode that only has one Jakarta Data provider.  The provider SPI here coordinates amongst all Jakarta Data providers and makes it work with no collisions on a single CDI extension, supplied by the Jakarta EE product/runtime.  The Jakarta EE product/runtime provides the CDI-centric solution when all of the Jakarta Data providers themselves cnanot.  It uses CDI events like `AfterBeanDiscovery` just like you are saying.  It's just the Jakarta EE product/runtime handling the CDI-centric aspect.  I don't see why this is being disputed so much.  Hopefully it is just a misunderstanding.